### PR TITLE
Creating docker build template

### PR DESCRIPTION
**Pull request by dineshgopagani**
## Summary
Creating an action template for docker build image.

## Ticket(s)
- [EAIP-356](https://gohypergiant.atlassian.net/browse/EAIP-356)

### CODE REVIEW

**Review by bbriggs — CHANGES_REQUESTED** [review]
Use this for referenec
https://github.com/bytebot-chat/gateway-irc/blob/main/.github/workflows/docker_build.yaml

**Comment by dineshgopagani** [comment]
> Use this for referenec
> https://github.com/bytebot-chat/gateway-irc/blob/main/.github/workflows/docker_build.yaml

Updated with Requested Changes. 
This template is to just build the docker images.

**Review by bbriggs — CHANGES_REQUESTED** [review]

File: workflow-templates/docker-build-template.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        with:
+          images: dkr.fraq.dev/bytebot/gateway-irc
+          tags: |
+            type=edge
+            type=ref,event=branch
+            type=ref,event=pr
+            type=sha,prefix=sha-,suffix=-${{env.NOW}},format=short
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+      -
```
[the remark this answers is not in the document]
Comment (dineshgopagani) [reply workflow-templates/docker-build-template.yml:37]: @bbriggs we don't need ECR login, as part of this template we are not pushing the image. We only need to verify if the docker image is build or not.

File: workflow-templates/docker-build-template.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        run: echo "NOW=$(date +%s)" >> $GITHUB_ENV
+      -
+        name: Echo current date
+        run: echo $NOW
+      -
+        name: Docker meta
+        id: meta
+        uses: crazy-max/ghaction-docker-meta@v2
+        with:
+          images: dkr.fraq.dev/bytebot/gateway-irc
```
[the remark this answers is not in the document]
Comment (dineshgopagani) [reply workflow-templates/docker-build-template.yml]: Updated.

File: workflow-templates/docker-build-template.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dineshgopagani) [reply workflow-templates/docker-build-template.yml:57]: Added ECR login

File: workflow-templates/docker-build-template.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
+name: Docker Build and Push
+
+on:
+  push:
+    branches:
+      - 'main'
```
Comment (bbriggs) [inline workflow-templates/docker-build-template.yml:6]: `stable` is the default branch at HG
```suggestion
      - 'stable'
```

File: workflow-templates/docker-build-template.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    tags:
+      - '*.*.*'
+  pull_request:
+    branches:
+      - 'main'
```
Comment (bbriggs) [inline workflow-templates/docker-build-template.yml:11]:
```suggestion
      - 'stable'
```

**Review by bbriggs — CHANGES_REQUESTED** [review]
Update branch name then LGTM

File: workflow-templates/docker-build-template.yml (hunk 3 above)
Comment (dineshgopagani) [reply workflow-templates/docker-build-template.yml]: Updated branch name to stable

File: workflow-templates/docker-build-template.yml (hunk 4 above)
Comment (dineshgopagani) [reply workflow-templates/docker-build-template.yml]: Updated branch name to stable

**Review by bbriggs — APPROVED** [review]

**Pull request merged**